services/QuillConnect/app/actions/fillInBlank.js

```diff
@@ -58,9 +58,11 @@ const actions = {
         if (error) {
           dispatch({ type: C.DISPLAY_ERROR, error: `Submission failed! ${error}`, });
         } else {
-          response.questionUID = newRef.key;
-          response.gradeIndex = `human${newRef.key}`;
-          dispatch(submitResponse(response));
+          if (response.text) {
```

Comment:
I note that we display a "Submission successfully saved!" message even in cases where this block is skipped, which might be a bit confusing.  But, more importantly, since we're not really expecting different front-end behavior based on whether there's text or not, it seems to me that this might be better implemented on the back-end.  That way you could implement it in (I would think) just one place, instead of six or seven.

Any thoughts on that approach?  It's possible that my assumption that it'd be easier on the backend is wrong, so I'm totally open to discussion here.

Reply:
Yeah, you're right. I think my old firebase mentality has the better of me here - I'm used to just doing things on the front end. I'll make a change to the CMS instead.